src/finance/PoolExposureSteward.sol

```diff
@@ -0,0 +1,220 @@
+ *
+ * -- Security Considerations
+ *
+ * -- Pools
+ * The pools managed by the steward are allowListed.
+ * As v2 pools are deprecated, the steward only implements withdrawals from v2.
+ *
+ * -- Permissions
+ * The contract implements OwnableWithGuardian.
+ * The owner will always be the respective network Short Executor (governance).
+ * The guardian role will be given to a Financial Service provider of the DAO.
+ *
+ * While the permitted Service Provider will have full control over the funds, the allowed actions are limited by the contract itself.
+ * All token interactions start and end on the Collector, so no funds ever leave the DAO possession at any point in time.
+ */
+contract PoolExposureSteward is OwnableWithGuardian, Rescuable, IPoolExposureSteward {
+  using CU for ICollector;
+  using CU for CU.IOInput;
+
+  /// @inheritdoc IPoolExposureSteward
+  ICollector public immutable COLLECTOR;
+
+  /// @inheritdoc IPoolExposureSteward
+  mapping(address poolV3 => bool) public poolsV3;
+  mapping(address poolV2 => bool) public poolsV2;
+
+  constructor(
+    address _owner,
+    address _guardian,
+    address collector,
+    address[] memory _poolsV2,
+    address[] memory _poolsV3
+  ) {
+    _transferOwnership(_owner);
+    _updateGuardian(_guardian);
+
+    COLLECTOR = ICollector(collector);
+
+    for (uint256 i; i < _poolsV2.length; i++) {
+      _approvePool(_poolsV2[i], false);
+    }
+
+    for (uint256 i; i < _poolsV3.length; i++) {
+      _approvePool(_poolsV3[i], true);
+    }
+  }
+
+  /// Steward Actions
+
+  /// @inheritdoc IPoolExposureSteward
+  function depositV3(address pool, address reserve, uint256 amount) external onlyOwnerOrGuardian {
+    _validateV3Pool(pool);
+
+    CU.depositToV3(COLLECTOR, CU.IOInput({pool: pool, underlying: reserve, amount: amount}));
+  }
+
+  /// @inheritdoc IPoolExposureSteward
+  function withdrawV3(address pool, address reserve, uint256 amount) external onlyOwnerOrGuardian {
+    _validateV3Pool(pool);
+
+    CU.IOInput memory withdrawData = CU.IOInput(pool, reserve, amount);
+    CU.withdrawFromV3(COLLECTOR, withdrawData, address(COLLECTOR));
+  }
+
+  /// @inheritdoc IPoolExposureSteward
+  function withdrawV2(address pool, address underlying, uint256 amount) external onlyOwnerOrGuardian {
+    _validateV2Pool(pool);
+
+    CU.withdrawFromV2(COLLECTOR, CU.IOInput(pool, underlying, amount), address(COLLECTOR));
+  }
+
+  /// @inheritdoc IPoolExposureSteward
+  function migrateV2toV3(address v2Pool, address v3Pool, address underlying, uint256 amount)
+    external
+    onlyOwnerOrGuardian
+  {
+    _validateV2Pool(v2Pool);
+    _validateV3Pool(v3Pool);
+    _migrateV2toV3(v2Pool, v3Pool, underlying, amount);
+  }
+
+  /// @inheritdoc IPoolExposureSteward
+  function migrateV2toV3(address v2Pool, address v3Pool, address[] calldata underlyings, uint256[] calldata amounts)
```

Comment:
It's a bit weird that you can migrate a batch, but you cannot batch a withdrawal or deposit.

I'd suggest to remove the batch overload and just add Multicall, like we did here: https://github.com/bgd-labs/aave-stewards/blob/main/src/maintenance/ClinicSteward.sol#L50